extension/src/bs/JdtlsImporterConnector.ts

```diff
@@ -0,0 +1,52 @@
+import * as net from "net";
+import * as rpc from "vscode-jsonrpc/node";
+import * as vscode from "vscode";
+import * as path from "path";
+
+export const GET_IMPORTER_PIPE_NAME = "gradle.getImporterPipeName";
+
+export class JdtlsImporterConnector {
+    private importerConnection: rpc.MessageConnection | null = null;
+    private importerPipeServer: net.Server;
+    private importerPipePath: string;
+    private readonly context: vscode.ExtensionContext;
+
+    constructor(context: vscode.ExtensionContext) {
+        this.context = context;
+    }
+
+    //receive the pipe name from Java jdt.ls importer
+    public async waitForImporterPipePath(): Promise<void> {
+        return new Promise((resolve) => {
+            this.context.subscriptions.push(
+                vscode.commands.registerCommand(GET_IMPORTER_PIPE_NAME, (pipeName: string) => {
+                    this.importerPipePath = path.resolve(pipeName);
```

Comment:
This binds the command to this obj, which looks a little bit wired.

I would suggest changing the method to emit an event, And resolve the problem when that event comes.